client/components/DisplayCalendar/Dates/Dates.js

```diff
@@ -18,252 +18,202 @@ class Dates extends React.Component {
-              );
-              datesInMonth++;
-              continue;
-            }
-
-            allDates[date] = (
+          if (nullDaysTracker < dayOfWeekStart) {
+            allDatesInMonth[nullDaysTracker + 7] = (
               <button
-                disabled={false}
-                key={date - dayOfWeekForButtonsIndex - 1}
-                name={new Date(
-                  currentYear,
-                  currentMonthIndex,
-                  date - dayOfWeekForButtonsIndex - 1
-                ).toLocaleDateString()}
-                onClick={selectDate}
-              >
-                {date - dayOfWeekForButtonsIndex - 1}
-              </button>
+                key={
+                  weekdayNames[dayOfWeekStart] +
+                  '-nulled-' +
+                  months[currentMonthIndex] +
+                  '-' +
+                  nullDaysTracker +
+                  '-'
+                }
+                disabled={true}
+              ></button>
             );
+            nullDaysTracker++;
+            eachDate--;
+            continue;
           }
-
-          combinedCalendars.push(
-            <div
-              className="full-month-container"
-              key={months[currentMonthIndex]}
+          allDatesInMonth[dayOfWeekStart + eachDate + 6] = (
+            <button
+              key={new Date(
+                currentYear,
+                currentMonthIndex,
+                eachDate
+              ).toLocaleDateString()}
+              name={new Date(
+                currentYear,
+                currentMonthIndex,
+                eachDate
+              ).toLocaleDateString()}
+              onClick={selectDate}
             >
-              {[
-                <div
-                  className="month-name-container"
-                  key={months[currentMonthIndex]}
-                >
-                  {months[currentMonthIndex]}
-                </div>,
-                <div
-                  className="weekday-names-container"
-                  key={months[currentMonthIndex] + 'weekdays'}
-                >
-                  {weekdayNameDivs}
-                </div>,
-                <div
-                  className="date-container"
-                  key={months[currentMonthIndex] + 'dates'}
-                >
-                  {allDates}
-                </div>,
-              ]}
-            </div>
+              {eachDate}
+            </button>
           );
         }
+      }
 
+      determineMonthStartDayOfWeek(howManyDaysInMonth);
+
+      return (
+        <div
+          className="weekdays-dates-container"
+          key={months[currentMonthIndex] + ' weeknames '}
+        >
+          {allDatesInMonth}
+        </div>
+      ); // arr
+    };
+
+    // END HELPER FUNCTIONS
+
+    weekdayNames.forEach((day, index) =>
+      weekdayNameDivs.push(<div key={index}>{day}</div>)
+    );
+
+    // returns current month as only option to book
+    if (monthsInAdvance === 0) {
```

Comment:
i like the refactor. this feels much easier/cleaner to understand how youre creating the combinedCalendars